server/restricted/restrictedMiddleware.ts

```diff
@@ -24,7 +26,9 @@ function validateToken(req: any, res: any, next: any) {
    }
 }
 
-function checkForRole(req: any, res: any, next: any) {
+// Only admins is able to make CRUD operations to products and
+// able to make users admins
+function checkForRole(req: any, res: Response, next: NextFunction) {
```

Comment:
Since I added ```decodedToken``` to the headers whenever I add the ```Request type``` it yells at me about decodedToken doesn't exist in the ```req```
I'll look more into this later.